File: 백준/Bronze/11050. 이항 계수 1/이항 계수 1.cs
using System;
using System.Text;

class Program
{
    public static void Main(string[] args)
    {
        string[] input = Console.ReadLine().Split(' ');

        int N = int.Parse(input[0]);
        int K = int.Parse(input[1]);

        int result = 1;

        if (K != 0 || K != N)
        {
            for (int i = 0; i < K; i++)
            {
                result *= (N - i);
            }

            for (int i = 0; i < K; i++)
            {
                result /= (K - i);
            }
        }

        Console.WriteLine(result);
    }
}

File: 백준/Bronze/11050. 이항 계수 1/이항 계수 1.java
import java.io.*;

class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int N = Integer.parseInt(input[0]);
        int K = Integer.parseInt(input[1]);

        System.out.println(BC(N, K));
    }

    public static int BC(int N, int K){
        if(N == K || K == 0){
            return 1;
        }

        return BC(N - 1, K - 1) + BC(N - 1, K);
    }
}